docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,906 @@ aliases:
+project: Because the project has a Git repository associated with it, Brigade
+is automatically fetching a clone of that repository and attaching it to the
+Worker in charge of running the script.
 
-Using groups, you can create sophisticated pipelines.
+By default, the repository contents are not automatically mounted to jobs in
+the project's Brigade script. However, mounting the contents to a job is easily
+accomplished via the `sourceMountPath` configuration on a job's
+`primaryContainer`.
 
-Sometimes you may want to declare groups ahead of time and then run them, much as you do
-with jobs. This is an alternative to using the Group static methods.
+The following example shows how a job can be configured to access the repo in
+order to run a test target. It also configures `workingDirectory` with the same
+value as `sourceMountPath` so that the job needn't worry about changing into
+the appropriate directory before running commands:
 
 ```javascript
-const { events, Job, Group } = require("brigadier")
-
-events.on("exec", () => {
-  var hello = new Job("hello", "alpine:3.4", ["echo hello"])
-  var goodbye = new Job("goodbye", "alpine:3.4", ["echo goodbye"])
-
-  var helloAgain = new Job("hello-again", "alpine:3.4", ["echo hello again"])
-  var goodbyeAgain = new Job("bye-again", "alpine:3.4", ["echo bye again"])
+const { events, Job } = require("@brigadecore/brigadier");
+const localPath = "/workspaces/brigade";
 
-
-  var first = new Group()
-  first.add(hello)
-  first.add(goodbye)
-
-  var second = new Group()
-  second.add(helloAgain)
-  second.add(goodbyeAgain)
-
-  first.runAll().then( () => second.runAll() )
+events.on("brigade.sh/github", "push", async event => {
+  let test = new Job("test", "debian", event);
+  test.primaryContainer.sourceMountPath = localPath;
+  test.primaryContainer.workingDirectory = localPath;
+  test.primaryContainer.command = ["bash"];
+  test.primaryContainer.arguments = ["-c", "make test"];
+  await test.run();
 })
-```
-[brigade-12.js](../../examples/brigade-12.js)
-
-The above creates two groups, and then later executes them. The order of execution would
-be:
-
-- Run both of the jobs in the `first` group.
-- Once those two jobs have both completed, run both of the jobs in the `second` group.
 
-This is the way groups can be used to control the order in which groups of jobs are run.
+events.process();
+```
 
-So far we have looked at the brigade.js file, the event registry, and jobs and groups. As we
-advance our way through Brigade, we will next take a look at how `brigade.js` scripts can work
-with the BrigadeEvent and Project objects that are sent to every event handler.
+Being able to associated a Git repository to a project is a convenient way to
+provide version-controlled data to our Brigade scripts. And it makes Brigade a
+great tool for executing CI pipelines, deployments, packaging tasks, end-to-end
+tests, and other DevOps tasks.
```

Comment:
I think I might add that even for use cases not otherwise requiring git, storing scripts and other configuration in a git repo is a recommended alternative to embedding complex scripts and configuration in the project definition.